### PR TITLE
[risk=low][RW-10991] Runtime-utils unit tests

### PR DESCRIPTION
**Pull request by jmthibault79**
Tests for most of runtime-utils... after splitting off hooks and associated functions to a new runtime-hooks file.  Continues #8143

No functional changes expected.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/utils/runtime-utils.tsx [hunk 1]
```diff
@@ -209,17 +197,22 @@ export const diffsToUpdateMessaging = (
...
-  if (diffType === AnalysisDiffState.NEEDS_DELETE) {
-    if (diskDiffType === AnalysisDiffState.NEEDS_DELETE) {
-      return recreateEnvAndPDUpdate;
-    }
-    return recreateEnvUpdate;
-  } else if (diffType === AnalysisDiffState.CAN_UPDATE_WITH_REBOOT) {
-    return rebootUpdate;
-  } else {
+
+  return cond(
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:201]: same logic

File: ui/src/app/utils/runtime-utils.tsx [hunk 2]
```diff
@@ -574,30 +567,33 @@ const presetEquals = (a: Runtime, b: Runtime): boolean => {
 
 export const fromAnalysisConfig = (analysisConfig: AnalysisConfig): Runtime => {
   const {
+    autopauseThreshold,
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:570]: same logic

File: ui/src/app/utils/runtime-utils.tsx [hunk 3]
```diff
@@ -631,20 +627,6 @@ const canUseExistingDisk = (
   );
 };
 
-const diskNeedsSizeIncrease = (
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:634]: moved to runtime-hooks because it's only used there

File: ui/src/app/utils/runtime-utils.tsx [hunk 4]
```diff
@@ -743,7 +727,7 @@ export const toAnalysisConfig = (
   runtime: Runtime,
   existingDisk: Disk | null
 ): AnalysisConfig => {
-  if (runtime.gceConfig) {
+  const toGceConfig = () => {
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:730]: same logic

File: ui/src/app/utils/runtime-utils.spec.tsx [hunk 5]
```diff
@@ -1159,8 +1180,173 @@ describe(withAnalysisConfigDefaults.name, () => {
...
+  it('returns true when a larger disk than the existingDisk is chosen', () => {
+    const existingDisk = stubDisk();
+    const { diskType: detachableType } = existingDisk;
+    const size = existingDisk.size + 1;
+    expect(
+      canUseExistingDisk({ detachableType, size }, existingDisk)
+    ).toBeTruthy();
+  });
+
+  it('returns flase when a smaller disk than the existingDisk is chosen', () => {
```
Comment (evrii) [inline ui/src/app/utils/runtime-utils.spec.tsx:1329]: Small typo here

**Review by evrii — APPROVED** [review]
This looks good to me. Thanks for taking the time to simplify this. The unit tests helped me to have a better idea how the underlying code is working.

**Pull request merged**